src/main/java/org/folio/rest/validator/ResourcePutBodyValidator.java

```diff
@@ -0,0 +1,91 @@
+package org.folio.rest.validator;
+
+import java.util.List;
+import java.util.Objects;
+
+import org.apache.commons.lang3.StringUtils;
+import org.folio.rest.exception.InputValidationException;
+import org.folio.rest.jaxrs.model.Coverage;
+import org.folio.rest.jaxrs.model.EmbargoPeriod.EmbargoUnit;
+import org.folio.rest.jaxrs.model.ResourceDataAttributes;
+import org.folio.rest.jaxrs.model.ResourcePutRequest;
+
+public class ResourcePutBodyValidator {
+  
+  private static final String INVALID_REQUEST_BODY_TITLE = "Invalid request body";
+  private static final String INVALID_REQUEST_BODY_DETAILS = "Json body must contain data.attributes";
+  private static final String INVALID_IS_SELECTED_TITLE = "Resource cannot be updated unless added to holdings";
+  private static final String INVALID_IS_SELECTED_DETAILS = "Resource must be added to holdings to be able to update";
+  
+  public void validate(ResourcePutRequest request, boolean isTitleCustom) {
+    if (request == null ||
+      request.getData() == null ||
+      request.getData().getAttributes() == null) {
+      throw new InputValidationException(INVALID_REQUEST_BODY_TITLE, INVALID_REQUEST_BODY_DETAILS);
+    }
+    
+    ResourceDataAttributes attributes = request.getData().getAttributes();
+    boolean isSelected = attributes.getIsSelected().booleanValue();
+    String cvgStmt = attributes.getCoverageStatement();
+
+    /**
+     * Updates cannot be made to a resource unless it is selected
+     */
+    if(isSelected) {
+      /**
+       * Following fields can be updated only for a custom resource although UI sends complete payload
+       * for both managed and custom resources
+       */
+      if(isTitleCustom) {
+        String name = attributes.getName();
+        String pubType = attributes.getPublicationType() != null ? attributes.getPublicationType().value() : null;
+        String pubName = attributes.getPublisherName();
+        String edition = attributes.getEdition();
+        String description = attributes.getDescription();
+        String url = attributes.getUrl();
+        
+        ValidatorUtil.checkIsBlank("name", name);
+        ValidatorUtil.checkMaxLength("name", name, 400);
+        ValidatorUtil.checkIsBlank("publicationType", pubType);
+        if(!StringUtils.isBlank(pubName)) {
+          ValidatorUtil.checkMaxLength("publisherName", pubName, 250);
+        }
+        if(!StringUtils.isBlank(edition)) {
+          ValidatorUtil.checkMaxLength("edition", edition, 250);
+        }
+        if(!StringUtils.isBlank(description)) {
+          ValidatorUtil.checkMaxLength("description", description, 1500);
+        }
+        if(!StringUtils.isBlank(url)) {
+          ValidatorUtil.checkMaxLength("url", url, 600);
+          ValidatorUtil.checkUrlFormat("url", url);
+        }
+        if(Objects.nonNull(attributes.getIdentifiers())) {
```

Comment:
I just checked that if array is null in json then it defaults to empty list, so there's no need to check for null here and in a few places below:
Objects.nonNull(attributes.getCustomCoverages()
!Objects.isNull(customCoverages)